vmngclient/api/tenant_api.py

```diff
@@ -0,0 +1,108 @@
+from typing import List, Optional
+
+from vmngclient.api.basic_api import Device
+from vmngclient.dataclasses import TenantInfo, TierInfo
+from vmngclient.session import vManageSession
+from vmngclient.utils.creation_tools import create_dataclass
+
+
+class TierNameNotFoundError(Exception):
+    pass
+
+
+class TenantNameNotFoundError(Exception):
+    pass
+
+
+class InvalidOperationError(Exception):
+    """The exception that is thrown when a method call is invalid for the object's current state."""
+
+    def __init__(self, message: str = "The input sequence contains more than one element."):
+        self.message = message
+        super().__init__()
+
+
+# TODO tests
+class TenantsAPI:
+    def __init__(self, session: vManageSession):
+        self.session = session
+
+    def get_tenants(
+        self, device_id: Optional[Device] = None, name: Optional[str] = None, organization_name: Optional[str] = None
+    ) -> List[TenantInfo]:
+        """Lists all the tenants on the vManage.
+
+        In a multitenant vManage system, this API is only avaiable in the Provider view.
+
+        Args:
+            device_id: Lists all tenants associated with a vSmart.
+
+        Returns:
+            List[TenantInfo]
+        """
+
+        if device_id:
+            raise NotImplementedError()
+
+        response = self.session.get_data("/dataservice/tenant")
+        tenants = [create_dataclass(TenantInfo, tenant_info) for tenant_info in response]
+        if any([name, organization_name]) is False:
+            return tenants
+
+        # TODO
```

Comment:
TODO is for filtering for both name and organization_name?

Reply:
yes. In my opinion its very ugly way for filtering objects in that way. Do not know the optimal solution yet.